src/components/Datatable/Datatable.tsx

```diff
@@ -172,20 +176,31 @@ const Datatable = <D extends Record<string, unknown>>({
     [onRowsSelect],
   );
 
+  const memoizedTableConfig = useDeepCompareMemo(
+    () => ({
+      hasSelection,
+      onSelect: handleOnRowsSelect,
+      defaultPageSize,
+      ...restTableConfig,
+    }),
+    [defaultPageSize, handleOnRowsSelect, hasSelection, restTableConfig],
+  );
+
   return (
-    <DatatableContext.Provider
-      value={{
-        totalLength: totalDataSize,
-        selectedIds,
-        selectedLength: selectedIds.length,
-        defaultHiddenColumns,
-        defaultColumnOrder,
-        hasExclusionLogic,
-        setHasExclusionLogic,
-        hasSelection,
-      }}
-    >
-      <StyledDatatable flexDirection="column" {...props}>
+    <StyledDatatable flexDirection="column" {...props}>
+      <DatatableContext.Provider
+        value={{
+          totalLength: totalDataSize,
+          selectedIds,
+          selectedLength: selectedIds.length,
+          defaultHiddenColumns,
+          defaultColumnOrder,
+          hasExclusionLogic,
+          setHasExclusionLogic,
+          hasSelection,
+          setShouldResetSelectedRows,
```

Comment:
Pls, who is the consumer of `setShouldResetSelectedRows`?

Reply:
The consumer is `ElementCounter`, this handles situation when you have some selected items and choose `Select none` from the dropdown menu. It wasn't working before, now the `Table` component gets notified through this prop that it should reset selection

Reply:
Right, I missed `ElementCounter`, see it now. Thanks for explanation